pkgs/os-specific/bsd/freebsd/pkgs/locales.nix

```diff
@@ -0,0 +1,51 @@
+{
+  mkDerivation,
+  lib,
+  symlinkJoin,
+  bsdSetupHook,
+  freebsdSetupHook,
+  makeMinimal,
+  install,
+  tsort,
+  lorder,
+  mandoc,
+  groff,
+  localedef,
+  allLocales ? true,
+  locales ? [ "en_US.UTF-8" ],
+}:
+let
+  build =
+    name: needsLocaledef:
+    mkDerivation {
```

Comment:
Do we really need to use 11 derivations for this, rather than 1?

Reply:
This is without a doubt the most elegant way to tackle this build using the infrastructure we have set up for building packages out of the FreeBSD source tree. Each derivation is invoking a separate makefile in a separate directory and has a separate set of build inputs to ensure no cross-contamination.

The alternative would be basically overriding buildPhase and installPhase to build and install with several makefiles.

What is the downside of many small derivations, even if they're not user-facing?

Reply:
I indeed want to have a `libc0` so we can stop doing all the extra libs in `postInstall` for that too (future work I am planning on).